external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -136,6 +136,10 @@ pub struct DefLoc {
 }
 
 impl DefLoc {
+    pub fn new(fhash: FileHash, start: Position) -> Self {
```

Comment:
Also not strictly necessary (for the same reasons as described [here](https://github.com/MystenLabs/sui/pull/17900#discussion_r1612246278)) but using it instead of struct constructors made the code look nicer so I left it